src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -64,19 +70,27 @@ string filterbycollection
             }
             catch (ArgumentException ex)
             {
-                logger.LogError(ex.Message);
+#if DEBUG
+                logger.LogCritical(ex, ex.Message);
+#else
+                logger.LogCritical(ex.Message);
+#endif
                 return;
             }
             try
             {
                 if (output.Exists)
                 {
-                    throw new IOException("The file you're writing to already exists. Please input a new file path.");
+                    throw new IOException($"The file {output} already exists. Please input a new file path.");
                 }
             }
             catch (IOException ex)
```

Comment:
sorry, I missed something on my initial review. Instead of having a bunch of try catches in this method, move to a single one that encapsulates the whole body, it'll make the code much smaller, and easier to read

Reply:
Yeah that makes sense. Resolving..